src/test/java/org/folio/fqm/service/FqlToSqlConverterServiceTest.java

```diff
@@ -0,0 +1,301 @@
+        """
+          {"field1": {"$regex": "some_text"}}""",
+        condition("{0} ~* {1}", field("field1"), val("some_text"))
+      ),
+      Arguments.of(
+        "in list",
+        """
+          {"field1": {"$in": ["value1", 2, true]}}""",
+        or(
+          field("field1").equalIgnoreCase("value1"),
+          field("field1").eq(2),
+          field("field1").eq(true)
+        )
+      ),
+      Arguments.of(
+        "not in list",
+        """
+          {"field1": {"$nin": ["value1", 2, true]}}""",
+        and(
+          field("field1").notEqualIgnoreCase("value1"),
+          field("field1").notEqual(2),
+          field("field1").notEqual(true)
+        )
+      ),
+
+      Arguments.of(
+        "contains string",
+        """
+          {"arrayField": {"$contains": "some value"}}""",
+        field("arrayField").containsIgnoreCase("some value")
+      ),
+      Arguments.of(
+        "contains numeric",
+        """
+          {"arrayField": {"$contains": 10}}""",
+        field("arrayField").contains(10)
+      ),
+
+      Arguments.of(
+        "not contains string",
+        """
+          {"arrayField": {"$not_contains": "some value"}}""",
+        field("arrayField").notContainsIgnoreCase("some value").or(field("arrayField").isNull())
+      ),
+      Arguments.of(
+        "not contains numeric",
+        """
+          {"arrayField": {"$not_contains": 10}}""",
+        field("arrayField").notContains(10).or(field("arrayField").isNull())
+      ),
+
+      Arguments.of(
+        "complex condition",
+        """
+          {
+            "$and": [
+              {"field1": {"$eq": "some value"}},
+              {"field2": {"$eq": true}},
+              {"field3": {"$lte": 3}},
+              {"field1": {"$eq": false}},
+              {"field2": {"$in": ["value1", 2, true]}},
+              {"field3": {"$ne": 5}},
+              {"field1": {"$gt": 9}},
+              {"field2": {"$lt": 11}},
+              {"field3": {"$nin": ["value1", 2, true]}}
+            ]
+          }""",
+        field("field1").equalIgnoreCase("some value")
+          .and(field("field2").eq(true))
+          .and(field("field3").lessOrEqual(3))
+          .and(field("field1").eq(false))
+          .and(
+            or(
+              field("field2").equalIgnoreCase("value1"),
+              field("field2").eq(2),
+              field("field2").eq(true)
+            )
+          )
+          .and(field("field3").notEqual(5))
+          .and(field("field1").greaterThan(9))
+          .and(field("field2").lessThan(11))
+          .and(
+            and(
+              field("field3").notEqualIgnoreCase("value1"),
+              field("field3").notEqual(2),
+              field("field3").notEqual(true)
+            )
+          )
+      )
+    );
+  }
+
+  @ParameterizedTest
+  @MethodSource("jooqConditionsSource")
+  void shouldGetJooqConditionForFqlEqualsStringCondition(String label, String fqlCondition, Condition expectedCondition) {
```

Comment:
Very cool! The smallest nitpick ever: The test name could probably be something like shouldGetJooqConditionForFqlCondition now.